### PR TITLE
FT connector: bugs fixes. Improve deploy scripts

### PR DESCRIPTION
**Pull request by karim-en**
* Fix initing `_is_whitelist_mode_enabled`
* Fix variable naming
* Fix `ft_on_transfer` arguments
* Fix whitelist mode enum
* Fix deploy scripts
* Add `update_factory_address` method to the near token locker contract 
* Add scripts to deposit ft on eth side

### CODE REVIEW

File: erc20-bridge-token/contracts/ProofConsumer.sol [hunk 1]
```diff
@@ -21,17 +21,17 @@ contract ProofConsumer {
...
     constructor (
-        bytes memory _nearTokenFactory,
+        bytes memory _nearTokenLocker,
         INearProver _prover,
         uint64 _minBlockAcceptanceHeight
     )  {
         require(
-            _nearTokenFactory.length > 0,
+            _nearTokenLocker.length > 0,
             "Invalid Near Token Factory address"
```
Comment (sept-en) [inline erc20-bridge-token/contracts/ProofConsumer.sol:30]: As we are changing the naming, we should also change the error message 
```suggestion
            "Invalid Near Token Locker address"
```

File: token-locker/src/lib.rs [hunk 2]
```diff
@@ -50,8 +50,8 @@ enum StorageKey {
 pub struct Contract {
     /// The account of the prover that we can use to prove.
     pub prover_account: AccountId,
-    /// Ethereum address of the token bridge contract, in hex.
-    bridge_address: EthAddress,
+    /// Ethereum address of the token factory contract, in hex.
+    factory_address: EthAddress,
```
Comment (sept-en) [inline token-locker/src/lib.rs:54]: Let's call it `eth_factory_address` to make it more clear

File: token-locker/src/lib.rs [hunk 3]
```diff
@@ -170,10 +170,10 @@ impl Contract {
         let event = EthUnlockedEvent::from_log_entry_data(&proof.log_entry_data);
         assert_eq!(
             event.bridge_address,
-            self.bridge_address,
+            self.factory_address,
             "Event's address {} does not match locker address of this token {}",
             hex::encode(&event.bridge_address),
-            hex::encode(&self.bridge_address),
+            hex::encode(&self.factory_address),
         );
```
Comment (sept-en) [inline token-locker/src/lib.rs:177]: Let's rename the variable in the event too

**Review by sept-en — CHANGES_REQUESTED** [review]

File: erc20-bridge-token/contracts/ProofConsumer.sol (hunk 1 above)
Comment (karim-en) [reply erc20-bridge-token/contracts/ProofConsumer.sol]: Fixed c112a8c38f43c876660c52e06a601a6ee01281f6

File: token-locker/src/lib.rs (hunk 2 above)
Comment (karim-en) [reply token-locker/src/lib.rs]: Fixed c112a8c38f43c876660c52e06a601a6ee01281f6

File: token-locker/src/lib.rs (hunk 3 above)
Comment (karim-en) [reply token-locker/src/lib.rs]: Fixed c112a8c38f43c876660c52e06a601a6ee01281f6

**Review by sept-en — APPROVED** [review]

**Pull request merged**